To: vim_dev@googlegroups.com
Subject: Patch 7.3.421
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.421
Problem:    Get E832 when setting 'undofile' in vimrc and there is a file to
	    be edited on the command line. (Toothpik)
Solution:   Do not try reading the undo file for a file that wasn't loaded.
Files:	    src/option.c


*** ../vim-7.3.420/src/option.c	2012-01-10 22:26:12.000000000 +0100
--- src/option.c	2012-01-28 17:55:24.000000000 +0100
***************
*** 7527,7536 ****
  	{
  	    /* When 'undofile' is set globally: for every buffer, otherwise
  	     * only for the current buffer: Try to read in the undofile, if
! 	     * one exists and the buffer wasn't changed. */
  	    if ((curbuf == save_curbuf
  				|| (opt_flags & OPT_GLOBAL) || opt_flags == 0)
! 		    && !curbufIsChanged())
  	    {
  		u_compute_hash(hash);
  		u_read_undo(NULL, hash, curbuf->b_fname);
--- 7527,7537 ----
  	{
  	    /* When 'undofile' is set globally: for every buffer, otherwise
  	     * only for the current buffer: Try to read in the undofile, if
! 	     * one exists and the buffer wasn't changed and the buffer was
! 	     * loaded. */
  	    if ((curbuf == save_curbuf
  				|| (opt_flags & OPT_GLOBAL) || opt_flags == 0)
! 		    && !curbufIsChanged() && curbuf->b_ml.ml_mfp != NULL)
  	    {
  		u_compute_hash(hash);
  		u_read_undo(NULL, hash, curbuf->b_fname);
*** ../vim-7.3.420/src/version.c	2012-01-26 20:58:21.000000000 +0100
--- src/version.c	2012-01-28 18:03:09.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     421,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
4. Your eyeglasses have a web site burned in on them.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
